src/arithmetic/big_native.rs

```diff
@@ -44,8 +44,8 @@ impl ZeroizeBN for BigInt {
 
 impl zeroize::Zeroize for BigInt {
     fn zeroize(&mut self) {
```

Comment:
Turned out that @elichai was working on zeroizing problem, and he re-implemented this method in this way:

```rust
    fn zeroize(&mut self) {
        use core::sync::atomic;
        // Copy the inner so we can read the data inside
        let original = unsafe { ptr::read(self) };
        // Replace self with a zeroed integer.
        unsafe { ptr::write_volatile(self, Self::zero()) };
        let (mut sign, uint) = original.0.into_parts();
        // Zero out the temp sign in case it's a secret somehow
        unsafe { ptr::write_volatile(&mut sign, Sign::NoSign) };
        // zero out the bigint's data itself.
        // This is semi-UB because it's a repr(Rust) type, but because it's a single field we can assume it matches the wrapper.
        let mut data: Vec<usize> = unsafe { core::mem::transmute(uint) };
        atomic::compiler_fence(atomic::Ordering::SeqCst);
        data.zeroize();
    }
```

I find it more robust, it should actually zeroize heap data. What do you think, @Sky-Nik ?

Reply:
looking at it again, the zeroizing of the `Sign` is mostly useless because `into_parts` returns it by value, so it actually zeroizes whatever is on the stack of this function.
AFAIU there's no easy way to zeroize that part

Reply:
This zeroize implementation is definitely better, although it is actually `original.num.into_parts()` in this version. At the very least, it passes my unsafe tests that tried keeping track of the memory addresses. I would definitely add it to the PR, even with the given sign problems.

However, @DmytroTym may now want to check if we should add any `Zeroizing<BigInt>`. Previously we decided to not get into it until we understand what will happen to `BigInt` zeroization on the higher level, but now sounds like a better time.

Reply:
@survived actually, now it also makes sense to add zeroize on drop to num-bigint, should I do it? Last time I decided to wait until we figure out the zeroization itself, but now I'm reasonably happy with it.

Reply:
Yes, let's do zeroize on drop, it's a good idea! I think we should have both backends zeroized on drop.

Reply:
Btw could you delete Zeroize implementation for gmp bigint? It doesn't make sense since it's zeroed out on drop anyway

Reply:
I'm confused, why are we removing the gmp one? (also I think the gmp one wasn't correct either, it overwrote the mpz pointer, not the data behind it)

Reply:
I believe that [gmp internally zeroes Mpz on drop](https://github.com/ZenGo-X/rust-gmp/blob/master/src/mpz.rs#L174)